src/configuration-store/configuration-store.ts

```diff
@@ -31,6 +32,7 @@ export interface IConfigurationStore<T> {
 
 export interface ISyncStore<T> {
   get(key: string): T | null;
+  getAll(): Record<string, T>;
```

Comment:
minor suggestion: a more idiomatic naming in typescript land would be `entries`, I think. Not a big deal though